packages/types/src/metrics.ts

```diff
@@ -99,4 +99,16 @@ export interface Metrics {
    * @experimental This API is experimental and might have breaking changes in the future.
    */
   gauge(name: string, value: number, data?: MetricData): void;
+
+  /**
+   * Adds a timing metric.
+   * The metric is added as a distribution metric.
+   *
+   * You can either directly capture a numeric `value`, or wrap a callback function in `timing`.
+   * In the latter case, the duration of the callback execution will be captured as a span & a metric.
+   *
+   * @experimental This API is experimental and might have breaking changes in the future.
+   */
+  timing(name: string, value: number, unit?: DurationUnit, data?: Omit<MetricData, 'unit'>): void;
+  timing<T>(name: string, callback: () => T, unit?: DurationUnit, data?: Omit<MetricData, 'unit'>): T;
```

Comment:
Did you check whether this JSDoc shows up properly in VS Code. I believe the JS Doc here might not show up when you use the callback variant.

Reply:
I just checked, seems fine for me for both variants:

<img width="966" alt="Screenshot 2024-05-28 at 10 15 09" src="https://github.com/getsentry/sentry-javascript/assets/2411343/e822c758-2cdc-459d-8d6d-d7e49ed578e3">

<img width="966" alt="Screenshot 2024-05-28 at 10 15 13" src="https://github.com/getsentry/sentry-javascript/assets/2411343/dd13fabd-d335-474d-a19a-d1ce2c74e849">